run_tests.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+cargo check
+cargo test
+
+cargo check --no-default-features
+cargo test --no-default-features --lib
+
+cargo check --no-default-features --features="alloc"
+cargo test --no-default-features --features="alloc" --lib
```

Comment:
Maybe it would be easier to build with `--features std` and `--features alloc` but test the code only with `--features std`? It would have reduced a lot of `#[cfg(feature = "alloc")] #[cfg(test)]`.